### ISSUE TITLE
DetachedInstanceError: Parent instance... with Discount by minor (discount.productclassdiscounts)

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#708]
_Imported from trac ticket 708.
Created by **kbower**
Opened in trac: Feb 16, 2011, 7:17:29 AM
Last modified in trac: Feb 16, 2011, 9:59:35 PM_
***

Error while evaluating this:
discount.productclassdiscounts
Complaint that discount is detached
```python
    @rule_property() 
    def process_applies(self):
        """
        work out whether this discount applies to the orderdetail (voided, productclass, etc)
        *and also should set the self.percentage when applies == 'Y' * !
        should not make any other changes, since they will not be rolled back properly
        """
        applies = False
        discount = self.discount
        line = self.orderdetail
        if not line.empty and line.itemprice > 0:
            if discount.productclassoverride == u'Y':
                # find the line's productclassid in the list of discount's productclassdiscounts
                productclassdisc = obj_with_attr(discount.productclassdiscounts, 'productclassid', line.product.productclassid)
                if productclassdisc:
                    applies = True
                    # we'll make up the attribute 'productclassdiscount'
                    self.productclassdiscount = productclassdisc
                    self.percentage = productclassdisc.percentage
            else:  
                # productclassoverride == u'N'
                applies = True
                if discount.type in u'CP':
                    self.percentage = discount.amount
        self.applies = applies and u'Y' or u'N'
```

**Comment by kentbower** [comment]
_**kbower** trac comment on Feb 16, 2011, 9:59:35 PM:_

(Changeset [[1209]](../commit/36251424d9dd92741ef25d6b4c23b4023b4acc47))
 * Fix fatal error (DetachedInstanceError: Parent instance) when adding discount codes that are restricted by minor code (closes #708)

**Comment by kentbower** [comment]
_**kbower** trac status change on Feb 16, 2011, 9:59:35 PM:_ **closed**
